pytorch_lightning/trainer/data_loading.py

```diff
@@ -57,21 +64,97 @@ def _percent_range_check(self, name):
         if not 0. <= value <= 1.:
             raise ValueError(msg)
 
-    def init_train_dataloader(self, model):
+    def call_prepare_data(self, model):
+        """
+        Let model download the data on proc==0 only
+        :param model:
+        """
+        # download data on DDP+
+        if self.use_ddp or self.use_ddp2:
+            if self.proc_rank == 0:
+                model.prepare_data()
+
+            # all processes wait until data download has happened
+            dist.barrier()
+
+        # data download/load on TPU
+        elif self.use_tpu and XLA_AVAILABLE:
+            if self.tpu_local_core_rank == 0:
+                model.prepare_data()
+
+            # all processes wait until data download has happened
+            torch_xla.core.xla_model.rendezvous("pl.TrainerDataLoadingMixin.get_dataloaders")
+
+        else:
+            # regular download
+            model.prepare_data()
+
+    def auto_add_sampler(self, dataloader, train):
+        # do nothing when user gives a sampler
+        dl_args = {
+            'dataset': dataloader.dataset,
+            'batch_size': dataloader.batch_size,
+            'shuffle': False,
```

Comment:
what if a user wants to shuffle batches (when running on a single machine)? i see below that in certain cases you're re-setting this value to `False`, did you intend to have it set to `True` here?

Reply:
I would rather move the shuffle to arguments as the others are taken from `dataloader` and only this is fixed